sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisExternalCatalogSuite.scala

```diff
@@ -28,13 +29,28 @@ import org.apache.spark.sql.catalyst.expressions.{Alias, AttributeReference}
 import org.apache.spark.sql.catalyst.plans.logical.{LocalRelation, Project}
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
+import org.apache.spark.util.Utils
 
 class AnalysisExternalCatalogSuite extends AnalysisTest with Matchers {
+  var tempDir: File = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    tempDir = Utils.createTempDir()
+  }
+
+  override def afterEach: Unit = {
+    try {
+      Utils.deleteRecursively(tempDir)
+    } finally {
+      super.afterEach()
+    }
+  }
   private def getAnalyzer(externCatalog: ExternalCatalog): Analyzer = {
     val conf = new SQLConf()
     val catalog = new SessionCatalog(externCatalog, FunctionRegistry.builtin, conf)
     catalog.createDatabase(
-      CatalogDatabase("default", "", new URI("loc"), Map.empty),
+      CatalogDatabase("default", "", new URI(tempDir.toString), Map.empty),
```

Comment:
Actually, how about just using `withTempDir`?

Reply:
> Actually, how about just using `withTempDir`?

Updated the code